packages/mobile/src/send/utils.ts

```diff
@@ -188,34 +184,30 @@ export function* handleSendPaymentData(
   }
   yield put(storeLatestInRecents(recipient))
 
-  if (data.currencyCode) {
-    yield put(selectPreferredCurrency(data.currencyCode as LocalCurrencyCode))
-    const action: FetchCurrentRateSuccessAction | FetchCurrentRateFailureAction = yield take([
-      Actions.FETCH_CURRENT_RATE_SUCCESS,
-      Actions.FETCH_CURRENT_RATE_FAILURE,
-    ])
-    if (action.type === Actions.FETCH_CURRENT_RATE_FAILURE) {
-      yield put(showError(ErrorMessages.EXCHANGE_RATE_FAILED))
-      Logger.warn(TAG, '@handleSendPaymentData failed to fetch current rate')
-      return
-    }
-  }
-
-  if (data.amount) {
-    // TODO: integrate with SendConfirmation component
-    const exchangeRate = yield select(getLocalCurrencyExchangeRate)
-    const amount = convertLocalAmountToDollars(data.amount, exchangeRate)
-    if (!amount) {
+  if (data.amount && data.currencyCode) {
+    const currency = data.currencyCode as LocalCurrencyCode
+    const exchangeRate = yield call(fetchExchangeRate, currency)
+    const dollarAmount = convertLocalAmountToDollars(data.amount, exchangeRate)
+    if (!dollarAmount) {
       Logger.warn(TAG, '@handleSendPaymentData null amount')
       return
     }
     const transactionData: TransactionDataInput = {
       recipient,
-      amount,
+      amount: dollarAmount,
       reason: data.comment,
       type: TokenTransactionType.PayPrefill,
     }
-    navigate(Screens.SendConfirmation, { transactionData, isFromScan: true })
+    const currencyInfo: CurrencyInfo = {
+      currencyCode: currency,
+      exchangeRate,
+    }
+
+    navigate(Screens.SendConfirmation, {
+      transactionData,
+      isFromScan: true,
+      currencyInfo,
```

Comment:
I would find this more readable if currencyCode and exchangeRate were explicitly passed